src/applications/disability-benefits/all-claims/utils/on-submit/purge-toxic-exposure-data.js

```diff
@@ -52,19 +52,20 @@ const hasSelectedConditions = conditions => {
  * non-applicable data. Called by purgeToxicExposureData for each exposure type
  * (gulfWar1990, gulfWar2001, herbicide, otherExposures).
  *
- * Handles four main cleanup scenarios for each exposure type:
+ * Handles five main cleanup scenarios for each exposure type:
  * 1. Removes null exposure fields as orphaned data
  * 2. Removes orphaned details when main selection is missing
  * 3. Removes entire section when all values are false/none or section is empty
- * 4. Removes "other" fields when null, invalid, or user selected 'none'
+ * 4. Removes details for items where selection is not true (keeps false selections, removes their details)
+ * 5. Removes "other" fields when null, 'none' selected, no selections, or corresponding 'other' not true
```

Comment:
I don't think, based on my other feedback, that this description is entirely accurate.